### PR TITLE
[CLOUDGA-24851] Remove root node info from cluster creation

### PR DESCRIPTION
**Pull request by daniel-yb**
Tested by running: 
```
./ybm cluster create --cluster-name daniel-test-cli --credentials username=admin,password=P@ssword1 --cloud-provider GCP --cluster-tier Dedicated --region-info region=northamerica-northeast2,num-nodes=3,num-cores=2,disk-size-gb=100
```

This PR depends on https://github.com/yugabyte/yugabyte-cloud/pull/18615.

### CODE REVIEW

File: internal/client/client.go [hunk 1]
```diff
@@ -267,7 +258,7 @@ func (a *AuthApiClient) buildClusterSpec(cmd *cobra.Command, regionInfoList []ma
 	}
 	if cmd.Flags().Changed("preferred-region") {
 		preferredRegion, _ := cmd.Flags().GetString("preferred-region")
-		if clusterInfo.GetFaultTolerance() != ybmclient.ClusterFaultTolerance("REGION") {
+		if clusterInfo.GetFaultTolerance() != ("REGION") {
```
Comment (posriniv) [inline internal/client/client.go:261]: maybe the backets are required aroung "REGION"

File: internal/client/client.go [hunk 2]
```diff
@@ -57,9 +57,9 @@ func GetVersion() string {
 	return cliVersion
 }
 
-// NewAuthClient function is returning a new AuthApiClient Client
+// NewAuthApiClient NewAuthClient function is returning a new AuthApiClient Client
 func NewAuthApiClient() (*AuthApiClient, error) {
-	url, err := ParseURL(viper.GetString("host"))
+	parseURL, err := ParseURL(viper.GetString("host"))
```
Comment (posriniv) [inline internal/client/client.go:62]: This can simply be url right?

File: internal/client/client.go [hunk 3]
```diff
@@ -302,115 +293,78 @@ func (a *AuthApiClient) buildClusterSpec(cmd *cobra.Command, regionInfoList []ma
...
+			// Asymmetric node configurations are only allowed for geo-partitioned clusters.
+			logrus.Fatalln("Synchronous cluster regions must have identical node configurations")
 		}
+		currRegionNodeInfo = (&clusterRegionInfo[i]).NodeInfo.Get()
 	}
 
 	if cmd.Flags().Changed("default-region") {
 		defaultRegion, _ := cmd.Flags().GetString("default-region")
-		if clusterInfo.GetClusterType() != ybmclient.ClusterType("GEO_PARTITIONED") {
+		if clusterInfo.GetClusterType() != ("GEO_PARTITIONED") {
```
Comment (posriniv) [inline internal/client/client.go:367]: Again, maybe remove the braces?

**Review by posriniv — APPROVED** [review]
Just some minor comments.

File: internal/client/client.go (hunk 2 above)
Comment (daniel-yb) [reply internal/client/client.go:62]: Screenshot 2024-11-25 at 6 23 59 PM
my IDE warns me to rename it.

**Pull request merged**